File: src/utils/crypto.py
import os
import base64
from cryptography.fernet import Fernet
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC


_KEY = None


def _get_key() -> bytes:
    global _KEY
    if _KEY is not None:
        return _KEY
    salt = b"EyeForge_salt_2026"
    kdf = PBKDF2HMAC(algorithm=hashes.SHA256(), length=32, salt=salt, iterations=100000)
    _KEY = base64.urlsafe_b64encode(kdf.derive(b"EyeForge_secret"))
    return _KEY


def encrypt(plain: str) -> str:
    if not plain:
        return ""
    f = Fernet(_get_key())
    return f.encrypt(plain.encode("utf-8")).decode("utf-8")


def decrypt(token: str) -> str:
    if not token:
        return ""
    try:
        f = Fernet(_get_key())
        return f.decrypt(token.encode("utf-8")).decode("utf-8")
    except Exception:
        return token

File: src-rs/src/crypto.rs
use base64::engine::general_purpose::URL_SAFE;
use base64::Engine as _;
use pbkdf2::pbkdf2_hmac;
use sha2::Sha256;

const SALT: &[u8] = b"EyeForge_salt_2026";
const SECRET: &[u8] = b"EyeForge_secret";
const ITERATIONS: u32 = 100_000;

fn derived_key() -> String {
    let mut key = [0_u8; 32];
    pbkdf2_hmac::<Sha256>(SECRET, SALT, ITERATIONS, &mut key);
    URL_SAFE.encode(key)
}

pub fn encrypt(plain: &str) -> String {
    if plain.is_empty() {
        return String::new();
    }

    let key = derived_key();
    let fernet = fernet::Fernet::new(&key).expect("invalid Fernet key");
    fernet.encrypt(plain.as_bytes())
}

pub fn decrypt(token: &str) -> String {
    if token.is_empty() {
        return String::new();
    }

    let key = derived_key();
    let fernet = fernet::Fernet::new(&key).expect("invalid Fernet key");

    match fernet.decrypt(token) {
        Ok(bytes) => String::from_utf8(bytes).unwrap_or_else(|_| token.to_string()),
        Err(_) => token.to_string(),
    }
}
